plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -201,6 +215,9 @@ public ConnectorPageSource createPageSource(
                     hiveColumnHandles::add,
                     () -> missingColumnNames.add(column.getBaseColumnName()));
         }
+        if (split.getDeletionVector().isPresent() && !regularColumns.contains(rowIdColumnHandle())) {
+            hiveColumnHandles.add(PARQUET_ROW_INDEX_COLUMN);
```

Comment:
what if regularColumns contains mergeRowIdColumnHandle ?
we will have PARQUET_ROW_INDEX_COLUMN twice in hiveColumnHandles

Reply:
i realize we currently don't support writing to tables with deletion vectors (writer version 7 requirement), but we want that in the future, so deletion vectors and mergeRowIdColumnHandle need to be able to co-exist.

`RowPredicate` abstraction will not be able to work efficiently on mergeRowIdColumnHandle due to ColumnarRow overhead. Do we want to make `RowPredicate` batched (not per-position)?

Reply:
Can you share the way to support batched `RowPredicate`?

Reply:
just remove `RowPredicate.test` and implement RowPredicate.filterPage directly?

```java
public interface RowPredicate // TODO rename?
        extends Function<Page, Page> {} 
```
```java
public RowPredicate createPredicate(List<DeltaLakeColumnHandle> columns)
{
    int filePositionChannel = rowPositionChannel(columns);

    if (deletedRows.isEmpty()) { // TODO is this case possible?
        return page -> page;
    }
    return page -> {
        int positionCount = page.getPositionCount();
        int[] retained = new int[positionCount];
        int retainedCount = 0;
        Block block = page.getBlock(filePositionChannel);
        for (int position = 0; position < positionCount; position++) {
            long filePosition = BIGINT.getLong(block, position);
            if (!deletedRows.contains(filePosition)) {
                retained[retainedCount] = position;
                retainedCount++;
            }
        }
        if (retainedCount == positionCount) {
            return page;
        }
        return page.getPositions(retained, 0, retainedCount);
    };
}
```

Reply:
Thanks for providing the code. Updated.